include/ProxyWifi/ProxyWifiService.hpp

```diff
@@ -95,30 +87,86 @@ struct OnDisconnectionArgs {
     DOT11_SSID disconnectedNetwork{};
 };
 
-struct ProxyWifiCallbacks
+enum class OperationType
 {
-    /// @brief Callback executed on a connection event
-    /// If the `EventSource` is the `Host`, it means indicated host interface connected
-    /// If the `EventSource` is the `Guest`, it means the guest is connecting (the connection request will be
-    /// completed only after this callback returns)
-    std::function<void(EventSource, const OnConnectionArgs&)> OnConnection;
-
-    /// @brief Callback executed on a disconnection event
-    /// If the `EventSource` is the `Host`, it means indicated host interface disconnected
-    /// If the `EventSource` is the `Guest`, it means the guest is disconnecting (the disconnection request will be
-    /// completed only after this callback returns)
-    std::function<void(EventSource, const OnDisconnectionArgs&)> OnDisconnection;
-
-    /// @brief Callback executed on the guest connection request that will cause a host interface to connect
-    /// The callback is invoked when before interfaces are connected/disconnected on the host, and after the connection
-    /// attempt is completed, whether it succeeds or fails.
-    std::function<void(GuestConnectStatus)> OnGuestConnectRequestProgress;
-
-    /// @brief Callback providing a list of networks that will be simulated by the Wi-Fi proxy
-    /// They will be shown as open networks, and are considered as permanently connected for the purpose of notifications
-    std::function<std::vector<WifiNetworkInfo>()> ProvideFakeNetworks;
+    GuestDirected,
+    HostMirroring
 };
 
+// Temporarily disable the "unused argument warning"
+#pragma warning(push)
+#pragma warning(disable : 4100)
+class ProxyWifiObserver
+{
+public:
+    struct ConnectCompleteArgs {
+        GUID interfaceGuid;
+        DOT11_SSID ssid;
+        DOT11_AUTH_ALGORITHM authAlgo;
+    };
+    struct DisconnectCompleteArgs {
+        GUID interfaceGuid;
+        DOT11_SSID ssid;
+    };
+
+    /// @brief An host WiFi interface connected to a network
+    virtual void OnHostConnection(const ConnectCompleteArgs& connectionInfo) noexcept
```

Comment:
Comment out variable instead of disabling the warning with `#pragma` above?

```suggestion
    virtual void OnHostConnection(const ConnectCompleteArgs& /* connectionInfo */) noexcept
```